### PR TITLE
Great split

### LINKED ISSUE #240: [Internal] Reorganize Frameworks to extract the Stencil nodes ("The Great Split")
There is multiple aims in this issue we want to discuss:

## Organization

We'll be reorganising the `GenumKit.framework` to split it into:
* `StencilSwiftKit.framework`, containing only the Stencil nodes specific to SwiftGen/Sourcery and don't have their place in Stencil proper
* `SwiftGenFramework.framework` for the rest of the code currently in GenumKit responsible for parsing the various input files and generating the contexts accordingly

We'll probably move each of those two frameworks in their own repos and define dependencies properly, so that both `SwiftGen` and `Sourcery` can depend on `StencilSwiftKit` and that each of them have their own `podspec` and versionning life cycle.

## Cleaning up the Stencil Nodes & Filters in GenumKit

* [ ] What are the ones that should now **be integrated into @kylef 's Stencil** proper directly?
* [ ] What are the ones that are **no longer needed** since Stencil evolved (I'm looking at you, `JoinNode`)
* [ ] What are the ones that should be **moved into a future `StencilSwiftKit.framework`**, so that wet can share them with @krzysztofzablocki 's [Sourcery](https://github.com/krzysztofzablocki/Sourcery)?

See associated GitHub Project here: ["The Great Split"](https://github.com/AliSoftware/SwiftGen/projects/1)

### PR DESCRIPTION
**Pull request by djbe**
Closes #240

### CODE REVIEW

**Comment by djbe** [comment]
Almost done, need to:
- [ ] update enrich call to match changes in https://github.com/SwiftGen/StencilSwiftKit/pull/19/files
- [ ] fix versions for kits to 1.0.0

File: SwiftGen/main.swift [hunk 1]
```diff
@@ -45,6 +46,6 @@ let main = Group {
   $0.addCommand("fonts", "generate code for your UIFonts", fontsCommand)
 }
 
-let version = Bundle(for: GenumKit.GenumTemplate.self)
+let version = Bundle.main
   .infoDictionary?["CFBundleShortVersionString"] as? String ?? "0.0"
 main.run("SwiftGen v\(version)")
```
Comment (AliSoftware) [inline SwiftGen/main.swift]: Could be interesting to fetch versions for SwiftGenKit and StencilSwiftKit too so we list them all there?

Comment (djbe) [inline SwiftGen/main.swift]: Do we also list the stencil version?

Comment (AliSoftware) [inline SwiftGen/main.swift]: Good point

**Comment by NachoSoto** [comment]
Awesome 🎉 looking forward to having this in Homebrew.

**Comment by AliSoftware** [comment]
🎉 You can follow the status of https://github.com/Homebrew/homebrew-core/pull/10132 to check when 4.2 will be accepted in homebrew (should not take long, they're usually quite reactive once the CI has built the bottles)

**Pull request merged**